openedx/adg/admin_dashboard/admin_settings/helpers.py

```diff
@@ -28,16 +29,25 @@ def extract_form_error(form_errors):
     Returns:
         str: error
     """
-    if EMAIL in form_errors:
-        error = form_errors[EMAIL][0].message
-        return INVALID_EMAILS_ERROR if error == THIS_FIELD_IS_REQUIRED else error
+    field = ''
 
+    if EMAIL in form_errors:
+        field = EMAIL
     elif PERMISSION in form_errors:
-        error = form_errors[PERMISSION][0].message
-        return PERMISSION_REQUIRED_ERROR if error == THIS_FIELD_IS_REQUIRED else error
-
+        field = PERMISSION
     elif BUSINESS_UNIT in form_errors:
-        return form_errors[BUSINESS_UNIT][0].message
+        field = BUSINESS_UNIT
+
+    if not field:
+        return
+
+    error = form_errors[field][0].message
+
+    return (
+        EMPTY_FIELDS_ERROR if error == THIS_FIELD_IS_REQUIRED else (
+            INVALID_EMAILS_ERROR if error == INVALID_EMAIL_ERROR else error
+        )
+    )
```

Comment:
This is getting confusing and unreadable. Let's use some constants for these conditions or break them down!